images/spire/config/main.tf

```diff
@@ -11,9 +11,14 @@ variable "extra_packages" {
   description = "Additional packages to install."
   type        = list(string)
   default     = []
+
 }
 
-module "accts" { source = "../../../tflib/accts" }
+module "accts" {
+  source = "../../../tflib/accts"
+  # agent runs as root while others as non-root, this condition is for that
+  run-as = var.name == "agent" ? 0 : 65532
```

Comment:
Any reason why not make it a variable here?
For example,
```terraform
variable "run-as" {
  description = "The user with which this should run as"
}
```

Then this line would be:
```terraform
  run-as = var.run-as
```